crates/ruff/src/rules/pyupgrade/rules/unnecessary_class_parentheses.rs

```diff
@@ -0,0 +1,96 @@
+use std::ops::Add;
+
+use ruff_text_size::{TextRange, TextSize};
+use rustpython_parser::ast::{self};
+
+use ruff_diagnostics::{AlwaysAutofixableViolation, Diagnostic, Edit, Fix};
+use ruff_macros::{derive_message_formats, violation};
+
+use crate::checkers::ast::Checker;
+use crate::registry::AsRule;
+
+/// ## What it does
+/// Checks for class definitions that include unnecessary parentheses after
+/// the class name.
+///
+/// ## Why is this bad?
+/// If a class definition doesn't have any bases, the parentheses are
+/// unnecessary.
+///
+/// ## Examples
+/// ```python
+/// class Foo():
+///     ...
+/// ```
+///
+/// Use instead:
+/// ```python
+/// class Foo:
+///     ...
+/// ```
+#[violation]
+pub struct UnnecessaryClassParentheses;
+
+impl AlwaysAutofixableViolation for UnnecessaryClassParentheses {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Unnecessary parentheses after class definition")
+    }
+
+    fn autofix_title(&self) -> String {
+        "Remove parentheses".to_string()
+    }
+}
+
+/// UP039
+pub(crate) fn unnecessary_class_parentheses(checker: &mut Checker, class_def: &ast::StmtClassDef) {
+    if !class_def.bases.is_empty() || !class_def.keywords.is_empty() {
+        return;
+    }
+
+    let contents = checker.locator.slice(class_def.range);
+
+    // Find the open and closing parentheses between the class name and the colon, if they exist.
+    let mut depth = 0u32;
+    let mut start = None;
+    let mut end = None;
+    for (i, c) in contents.char_indices() {
+        match c {
+            '(' => {
+                if depth == 0 {
+                    start = Some(i);
+                }
+                depth = depth.saturating_add(1);
```

Comment:
In which case would there be more than one parentheses pair in a class without bases?